cmd/dra-example-kubeletplugin/driver.go

```diff
@@ -28,6 +28,8 @@ import (
 	nasclient "github.com/kubernetes-sigs/dra-example-driver/api/example.com/resource/gpu/nas/v1alpha1/client"
 )
 
+var _ drapbv1.NodeServer = &driver{}
```

Comment:
Is this necessary at all? The interface conformance is already checked when the driver is passed to kubelet-plugin helper code here: https://github.com/kubernetes-sigs/dra-example-driver/blob/794daff2849e6d3015f3fa1cbb6e01c675ec7e8f/cmd/dra-example-kubeletplugin/main.go#L223

Try renaming NodePrepareResource to nodePrepareResource, for instance, and compiler will fail with 
```
$ make build
GOOS=linux go build ./...
# github.com/kubernetes-sigs/dra-example-driver/cmd/dra-example-kubeletplugin
cmd/dra-example-kubeletplugin/main.go:224:3: cannot use driver (variable of type *driver) as "k8s.io/kubelet/pkg/apis/dra/v1alpha2".NodeServer value in argument to plugin.Start: *driver does not implement "k8s.io/kubelet/pkg/apis/dra/v1alpha2".NodeServer (missing method NodePrepareResource)
make: *** [Makefile:48: build] Error 1
```

Reply:
Same about the check in controller driver.

Reply:
Perhaps so.

Reply:
It's good Golang practice to have such interface assertions.

Reply:
https://github.com/kubernetes-sigs/dra-example-driver/pull/9#discussion_r1182350297 actually shows one reason for it: compilation fails in `cmd/dra-example-kubeletplugin/main.go` but the problem is in `[cmd/dra-example-kubeletplugin/driver.go](https://github.com/kubernetes-sigs/dra-example-driver/pull/9/files/9f59860f2f6b5e057c0d55394f9bca4c8b29c319#diff-37a4b723138c8e7d62ccb364c381c4965f48c5792835f756c1ad1b43b8f58116)`. If that was some separate package, that package itself might build although it's broken.

Reply:
That's understandable, I wouldn't even ask if it was under pkg/ .